component/src/ibc/ibc_token.rs

```diff
@@ -0,0 +1,58 @@
+use ibc::core::ics24_host::identifier::{ChannelId, PortId};
+use penumbra_crypto::asset;
+
+/// IBC token respresents a token that was created through IBC.
+pub struct IBCToken {
```

Comment:
This should be at least `#[derive(Clone, Debug)]` and also probably `PartialEq`, `Eq`?